dmaws/build.py

```diff
@@ -21,8 +23,24 @@ def run_git_cmd(args, cwd, stdout=None):
     )
 
 
-def get_application_name(cwd):
-    repo_url = get_repo_url(cwd)
+def clone_or_update(repo_url):
+    app_name = get_application_name_from_url(repo_url)
+    repository_path = os.path.join(REPOS_PATH,
+                                   'digitalmarketplace-{}'.format(app_name))
+    if not os.path.exists(REPOS_PATH):
+        os.mkdir(REPOS_PATH)
+    if not os.path.exists(repository_path):
+        run_git_cmd(['clone', repo_url], REPOS_PATH)
+    else:
+        run_git_cmd(['reset', '--hard', 'HEAD'], repository_path)
+        run_git_cmd(['checkout', 'master'], repository_path)
+        run_git_cmd(['clean', '-fdx'], repository_path)
+        run_git_cmd(['pull'], repository_path)
```

Comment:
This is probably safe enough for us, but one case where this won't work as expected is if origin master was force-pushed. How about `fetch` followed by `git reset --hard origin/master`?